cmd/analyzer/parse_args.go

```diff
@@ -189,11 +190,18 @@ func errorInErgs(args *InArgs, flagset *flag.FlagSet) error {
 	}
 	if _, ok := supportedAnalysisTypesMap[*args.AnalysisType]; !ok {
 		flagset.PrintDefaults()
-		return fmt.Errorf("wrong analysis type %s; must be one of: %s", *args.AnalysisType, strings.Join(supportedAnalysisTypesList, separator))
+		return fmt.Errorf("wrong analysis type '%s'; must be one of: '%s'",
+			*args.AnalysisType, strings.Join(supportedAnalysisTypesList, separator))
 	}
 	if !supportedOutputFormatsMap[*args.OutputFormat] {
 		flagset.PrintDefaults()
-		return fmt.Errorf("wrong output format %s; must be one of: %s", *args.OutputFormat, strings.Join(supportedOutputFormatsList, separator))
+		return fmt.Errorf("wrong output format '%s'; must be one of: '%s'",
+			*args.OutputFormat, strings.Join(supportedOutputFormatsList, separator))
+	}
+	if !slices.Contains(supportedAnalysisTypesMap[*args.AnalysisType], *args.OutputFormat) {
+		flagset.PrintDefaults()
+		return fmt.Errorf("wrong output format '%s' for analysis type '%s'; must be one of: %s",
+			*args.OutputFormat, *args.AnalysisType, strings.Join(supportedAnalysisTypesMap[*args.AnalysisType], separator))
 	}
 	if *args.OutputFormat == DEBUGFormat && *args.AnalysisType != allEndpoints {
```

Comment:
also this `if` can be deleted (`if *args.OutputFormat == DEBUGFormat && *args.AnalysisType != allEndpoints`)

Reply:
done